### PR TITLE
[master] - Development (Boilerplate)

### PR DESCRIPTION
**Pull request by leonkozlowski**
# betfund-bet365

## Installation

From source
```bash
$ git clone https://github.com/betfund/betfund-bet365.git
$ cd betfund-lines

$ python3.7 -m venv venv
$ pip install -e .

# For test dependencies
$ pip install -e ".[testing]"
```

## Usage
GET Request to Upcoming Events Endpoint
```python
from betfund_bet365 import Bet365

client = Bet365()
upcoming_events = client.upcoming_events(sport="49")

print(upcoming_events)
```

```bash
{
  "success": 1,
  "pager": {
    "page": 1,
    "per_page": 50,
    "total": 102
  },
  "results": [
    {
      "id": "88107197",
      "sport_id": "92",
      "time": "1586480400",
      "time_status": "0",
      "league": {
        "id": "10036652",
        "name": "Moscow Liga Pro"
      },
      "home": {
        "id": "10423098",
        "name": "Evgenii Kryuchkov"
      },
      "away": {
        "id": "10433218",
        "name": "Aik Lulikyan"
      },
      "ss": null,
      "our_event_id": "2297143",
      "updated_at": "1586478473"
    }
  ]
}
```

Access response objects with dot notation
```python
from betfund_bet365 import Bet365

client = Bet365()
upcoming_events = client.upcoming_events(sport_id="49")

print(upcoming_events.success)
```

```bash
1
```

Access of array type `results` objects
```python
from betfund_bet365 import Bet365

client = Bet365()
upcoming_events = client.upcoming_events(sport_id="49")

print(upcoming_events.events[0].id)
```

```bash
"88107197"
```

## Environment Variables

+ BET365_HOST
    + `$ export BET365_HOST=yourHost`

+ BET365_KEY
    + `$ export BET365_KEY=yourSecretKey`

## Testing
```bash
pip install -e ".[testing]"

# Test with pytest
make tests

# Lint with flake8
make flake

# Lint with pylint
make lint
```

### CODE REVIEW

File: betfund_bet365/__main__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        api_key="someKey"
+    )
+
+    upcoming_events = client.upcoming_events(sport_id=sport_id)
+
+    return upcoming_events
+
+
+if __name__ == '__main__':  # pragma: no cover
+    main(sport_id="92")
```
Comment (mitchbregs) [inline betfund_bet365/__main__.py]: Wanna drop this for `master`?

File: betfund_bet365/client/client.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+Responses are parsed into Facade Access objects (Base Bet365Response)
+"""
+from typing import Optional, Union
+from urllib.parse import urljoin
+
+import requests
+
+import betfund_bet365.response as facades
+
+from betfund_bet365.response import Bet365Response
```
Comment (mitchbregs) [inline betfund_bet365/client/client.py:23]: Need for this import `import betfund_bet365.response as facades`? Also line spacing.

File: betfund_bet365/client/client.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+            delegation = getattr(
+                facades, RESPONSE_OBJECT_FACTORY.get(url_extras)
+            )
+
+            # Load the desired `.json()` response to ResponseObject
+            delegate_object = delegation(response.json())
+
+        except AttributeError:
+            # fall back on `.json()` if facade does not exist
+            return response.json()
```
Comment (mitchbregs) [inline betfund_bet365/client/client.py:85]: I 👀 you

File: betfund_bet365/response/in_play_events.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    @property
+    def type(self) -> str:
+        """Access for `type`."""
+        return self.get("type")
+
+    # @property
+    # def mnemonic(self):
+    #     """Access for `mnemonic`."""
+    #     this = Bet365Mnemonic
+    #     return this.value.mnemonic
```
Comment (mitchbregs) [inline betfund_bet365/response/in_play_events.py]: Needed?

File: tests/unit/test_base_response.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+# class TestFiResultBase(TestCase):
+#     """Unit tests for FiResultBase Facade."""
+#
+#     def setUp(self) -> None:
+#         """Instantiate FiResultBase."""
+#
+#         self.test_client = FiResultBase(
+#             load_json("testData/upcoming_events_table_tennis.json")
+#         )
```
Comment (mitchbregs) [inline tests/unit/test_base_response.py]: Needed?

**Review by mitchbregs — APPROVED** [review]
All LGTM.  Just some minor cleanup and good for merging.  Discussed IRL.  You are the man. 🚀

File: tests/unit/test_base_response.py (hunk 5 above)
Comment (leonkozlowski) [reply tests/unit/test_base_response.py]: removed

File: betfund_bet365/response/in_play_events.py (hunk 4 above)
Comment (leonkozlowski) [reply betfund_bet365/response/in_play_events.py]: removed as well

File: betfund_bet365/client/client.py (hunk 2 above)
Comment (leonkozlowski) [reply betfund_bet365/client/client.py:23]: IRL discussed - needed for factory

File: betfund_bet365/__main__.py (hunk 1 above)
Comment (leonkozlowski) [reply betfund_bet365/__main__.py]: Done.

**Pull request merged**